services/wallet/thirdparty/activity/alchemy/client.go

```diff
@@ -20,6 +22,7 @@ const AlchemyID = "alchemy"
 type Client struct {
 	ethClientGetter  rpc.EthClientGetter
 	connectionStatus *connection.Status
+	persistence      *Persistence
```

Comment:
Ugh, I'd like client to just be a client, putting persistence here seems wrong. I get why you did it though, we don't expose the provider-specific types outside the client since we do the conversion to the common type here. I'm having doubts whether that is a good approach, wondering if the conversion from provider-specific types to common types should happen somewhere else (some wrapper class around thirdparty/clients that implements the common interfaces)